yt_dlp/extractor/safari.py

```diff
@@ -22,33 +18,26 @@ class SafariBaseIE(InfoExtractor):
 
     LOGGED_IN = False
 
-    def _perform_login(self, username, password):
-        _, urlh = self._download_webpage_handle(
-            'https://learning.oreilly.com/accounts/login-check/', None,
-            'Downloading login page')
+    def is_logged(self):
+        return self._download_json_handle(
+            'https://api.oreilly.com/api/v2/me/', None, 'Checking if logged in',
+            fatal=False, headers={'Accept': 'application/json'}) is not False
 
-        def is_logged(urlh):
-            return 'learning.oreilly.com/home/' in urlh.geturl()
+    def _initialize_pre_login(self):
+        self.LOGGED_IN = self.is_logged()
```

Comment:
```suggestion
    def is_logged_in(self):
        self.LOGGED_IN = bool(self._request_webpage(
            'https://api.oreilly.com/api/v2/me/', None, note='Checking if logged in',
            fatal=False, errnote=False, headers={'Accept': 'application/json'}))
        return self.LOGGED_IN

    def _initialize_pre_login(self):
        self.is_logged_in()
```

Reply:
Committed in https://github.com/yt-dlp/yt-dlp/pull/3536/commits/fca3950b88228f2e7d24040c84f9495817dd84a5
fixed up in https://github.com/yt-dlp/yt-dlp/pull/3536/commits/10a2af1d54895d1c73c72c12d74af1155ca66a78.